scripts/generator/graphql_generator/csharp/SDK/Android/AndroidNativeCheckout.cs.erb

```diff
@@ -60,27 +85,23 @@ namespace <%= namespace %>.SDK.Android {
             var requiresShipping = checkout.requiresShipping();
 
             #if !SHOPIFY_MONO_UNIT_TEST
-            const string target = "com.shopify.unity.buy.androidpay.AndroidPayCheckoutSession";
-            const bool testing = true; // TODO parametrize for 3rd-party devs
-            using (var androidPayCheckoutSession = new AndroidJavaObject(target, GlobalGameObject.Name, testing)) {
-                object[] args = {
-                     MerchantName,
-                     key,
-                     pricingLineItemsString,
-                     currencyCodeString,
-                     CountryCodeString,
-                     requiresShipping
-                };
-                if (androidPayCheckoutSession.Call<bool>("checkoutWithAndroidPay", args)) {
-                    if (AndroidPayEventBridge == null) {
-                        AndroidPayEventBridge = GlobalGameObject.AddComponent<AndroidPayEventReceiverBridge>();
-                        AndroidPayEventBridge.Receiver = this;
-                    }
-                } else {
-                    // TODO: Create more meaningful error?
-                    var error = new ShopifyError(ShopifyError.ErrorType.NativePaymentProcessingError, "Unable to create Android Pay session.");
-                    OnFailure(error);
+            object[] args = {
+                    MerchantName,
+                    key,
+                    pricingLineItemsString,
+                    currencyCodeString,
+                    CountryCodeString,
+                    requiresShipping
+            };
+            if (AndroidPayCheckoutSession.Call<bool>("checkoutWithAndroidPay", args)) {
+                if (AndroidPayEventBridge == null) {
+                    AndroidPayEventBridge = GlobalGameObject.AddComponent<AndroidPayEventReceiverBridge>();
+                    AndroidPayEventBridge.Receiver = this;
                 }
+            } else {
+                // TODO: Create more meaningful error?
```

Comment:
What kind of things could we tell the consumer about? Would this only fail because it's not enabled on their store? That would be a great thing to relay along if that's the case. We can also introduce additional error types if it makes sense here.

Reply:
Actually, this `else` will only run if the Java `checkoutWithAndroidPay()` method returns `false` due to invalid or missing arguments, or incorrect JSON representation. It's been like this since the beginning, but if it happens, it's more likely our SDK's fault rather than the game dev's. I honestly think we should just validate user input on the Unity side and omit this error propagation. If the execution falls in this `else` once, it will do it every time, since our SDK would have a bug in this case.

Reply:
https://github.com/Shopify/unity-buy-sdk/pull/360/commits/82e61e32dd4888f6e521453d458bb65e248727ae